### PR TITLE
Add tendermint core tests (part 1)

### PR DESCRIPTION
**Pull request by screwyprof**

### CODE REVIEW

File: consensus/tendermint/core/backlog_test.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+		}
+
+		proposalPayload, _ := Encode(proposal)
+
+		msg := &message{
+			Code: msgProposal,
+			Msg:  proposalPayload,
+		}
+
+		c.storeBacklog(msg, validator.New(common.HexToAddress("0x0987654321")))
```
Comment (JekaMas) [inline consensus/tendermint/core/backlog_test.go]: Could you help me with this piece of code? Where is an asset step?

Comment (screwyprof) [reply consensus/tendermint/core/backlog_test.go]: My bad, forgot to add the assert. Fixed.

File: consensus/tendermint/core/backlog_test.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+		}
+
+		proposal := &Proposal{
+			Round:         big.NewInt(1),
+			Height:        big.NewInt(2),
+			ValidRound:    big.NewInt(1),
+			ProposalBlock: types.NewBlockWithHeader(&types.Header{}),
+		}
+
+		proposalPayload, _ := Encode(proposal)
```
Comment (b00ris) [inline consensus/tendermint/core/backlog_test.go]: Why did you skip error check?

Comment (screwyprof) [reply consensus/tendermint/core/backlog_test.go]: Good catch, thanks. A copy-paste from old tests. Fixed.

**Pull request merged**